### DISCUSSION TITLE
Pylance: "connect" is not exported from module "oracledb"

### DISCUSSION THREAD

**Discussion by gme-gpl** [oracle/python-oracledb#168] [Q&A] [answered]
I write code same as documentation, but VS Code Pylance raise a problem: **"connect" is not exported from module "oracledb"**

the code is:
```python
import oracledb

conn = oracledb.connect(...)  # error is here
cursor = conn.cursor()  # the type of `cursor` variable is `Cursor`, that's right
```

So I change the code to:
```python
from oracledb.connection import connect as oracle_connect

conn = oracle_connect(...)  # that is ok
cursor = conn.cursor()  # but the type of `cursor` variable is `Any` in mypy
```

Now, I must write more code:
```python
from oracledb.connection import connect as oracle_connect
from oracledb.cursor import Cursor

conn = oracle_connect(...)  # that is ok
cursor = conn.cursor()  # the type of `cursor` variable is `Cursor`, that's right
```

I use Python 3.10.9, and the oracledb's version is 1.2.2.
Do I have some way to use oracledb with Pylance

**Comment by cjbj** [comment, marked as answer]
Update to python-oracledb 1.3.  There were some [improvements](https://python-oracledb.readthedocs.io/en/latest/release_notes.html#oracledb-1-3-0-march-2023) in this release precisely to help in IDEs: "Improved type annotations."

**Reply by gme-gpl** [reply]
Thanks for your reply. It works!
